### PR TITLE
ICU-21349 Fix ICU4J reciprocal unit conversions

### PR DESCRIPTION
**Pull request by hugovdm**
unitPreferencesTest.txt was added to this PR by "ant copy-cldr-testdata" provided with a CLDR_DIR having release-39-alpha0 checked out.

Contributes to: icu-units#82

##### Checklist

- [X] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21349
- [X] Updated PR title and link in previous line to include Issue number
- [X] Issue accepted
- [X] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Comment by pedberg-icu** [comment]
We might have a chicken-and-egg test failure here, where the failure in this PR would be addressed by having https://github.com/unicode-org/icu/pull/1551 merged, and the failure in https://github.com/unicode-org/icu/pull/1551 would be addressed by having this one merged.

**Comment by hugovdm** [comment]
@pedberg-icu Ah true - I forgot to test this and fix separately: I developed on top of your branch, then rebased, and forgot to make the necessary data and testing file changes. (I've done a couple PRs like that recently.)

I'll try to get this fixed a few hours from now when I can return to my computer.

**Review by sffc — DISMISSED** [review]
Looks plausible, if all the test cases are still passing, but I defer to Younies for the code review because he knows this part of the code better than me.

**Comment by sffc** [comment]
Actually, please fix the ICU4J tests that are failing on CI:

```
2021-02-05T17:38:34.1499196Z Testsuite: com.ibm.icu.dev.test.impl.UnitsTest
2021-02-05T17:38:34.1500583Z Tests run: 6, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.184 sec
2021-02-05T17:38:34.1501283Z 
2021-02-05T17:38:34.1502014Z Testcase: testUnitPreferencesWithCLDRTests took 0.141 sec
2021-02-05T17:38:34.1502467Z 	FAILED
2021-02-05T17:38:34.1504546Z Test failed: TestCase: consumption-inverse, default, 001; Input: 1.1E+8 meter-per-cubic-meter; Expected Values: [1.1], Expected Units: [MeasureUnitImpl [kilometer-per-centiliter]]; Got unexpected result: [0.9090909090909090909090909090909091 consumption-liter-per-100-kilometer]
2021-02-05T17:38:34.1506714Z junit.framework.AssertionFailedError: Test failed: TestCase: consumption-inverse, default, 001; Input: 1.1E+8 meter-per-cubic-meter; Expected Values: [1.1], Expected Units: [MeasureUnitImpl [kilometer-per-centiliter]]; Got unexpected result: [0.9090909090909090909090909090909091 consumption-liter-per-100-kilometer]
2021-02-05T17:38:34.1508276Z 	at com.ibm.icu.dev.test.impl.UnitsTest.testUnitPreferencesWithCLDRTests(UnitsTest.java:543)
2021-02-05T17:38:34.1508668Z 
2021-02-05T17:38:34.1509034Z Testcase: testConverterWithCLDRTests took 0.032 sec
2021-02-05T17:38:34.1509475Z Testcase: testConverter took 0.005 sec
2021-02-05T17:38:34.1509831Z 	FAILED
2021-02-05T17:38:34.1510788Z testConverter inverse: MeasureUnitImpl [mile-per-gallon] back to MeasureUnitImpl [cubic-meter-per-meter] expected:<2.1383143939394E-6> but was:<386494.27948206547>
2021-02-05T17:38:34.1512373Z junit.framework.AssertionFailedError: testConverter inverse: MeasureUnitImpl [mile-per-gallon] back to MeasureUnitImpl [cubic-meter-per-meter] expected:<2.1383143939394E-6> but was:<386494.27948206547>
2021-02-05T17:38:34.1513281Z 	at com.ibm.icu.dev.test.impl.UnitsTest.testConverter(UnitsTest.java:360)
2021-02-05T17:38:34.1513628Z 
2021-02-05T17:38:34.1514017Z Testcase: testExtractConvertibility took 0 sec
2021-02-05T17:38:34.1514667Z Testcase: testComplexUnitConverterSorting took 0.002 sec
2021-02-05T17:38:34.1515167Z Testcase: testComplexUnitsConverter took 0.002 sec
```

**Comment by hugovdm** [comment]
Fixed now.

On my system, I've also combined this with PR 1551 and run the three main test classes to check they pass, and a full "ant check" is still running.

(I don't know how I got myself under the impression I got everything working previously - I think in a rush I sometimes miss the "Failures: 1" in output lines like these:

    [junit] Tests run: 6, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.632 sec

...)

**Review by sffc — DISMISSED** [review]

File: icu4c/source/test/testdata/cldr/units/unitPreferencesTest.txt
Comment (hugovdm) [inline icu4c/source/test/testdata/cldr/units/unitPreferencesTest.txt:83]: This file was added to this PR by simply running the "ant copy-cldr-testdata" build target, with CLDR_DIR having release-39-alpha0 checked out.

File: icu4j/main/tests/core/src/com/ibm/icu/dev/data/cldr/units/unitPreferencesTest.txt
Comment (hugovdm) [inline icu4j/main/tests/core/src/com/ibm/icu/dev/data/cldr/units/unitPreferencesTest.txt:83]: This file was added to this PR by simply running the "ant copy-cldr-testdata" build target, with CLDR_DIR having release-39-alpha0 checked out.

**Comment by hugovdm** [comment]
Only appveyor left. Tests have all passed, and "ant check" passed on my system when combined with PR 1551.

Reviewer/approver: feel free to click the "rebase and merge" button! I'm off to bed.

**Review by pedberg-icu — APPROVED** [review]
These changes look good to me

**Pull request merged**
